pystiche/papers/ulyanov_et_al_2016/core.py

```diff
@@ -0,0 +1,192 @@
+        device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+        images = ulyanov_et_al_2016_images(download=False)
+        style_image = images[style].read(device=device)
+    else:
+        style_image = style
+        device = style_image.device
+
+    if transformer is None:
+        transformer = ulyanov_et_al_2016_transformer(
+            impl_params=impl_params,
+            instance_norm=instance_norm,
+            stylization=stylization,
+        )
+        transformer = transformer.train()
+    transformer = transformer.to(device)
+
+    if criterion is None:
+        criterion = ulyanov_et_al_2016_perceptual_loss(
+            impl_params=impl_params,
+            instance_norm=instance_norm,
+            stylization=stylization,
+        )
+        criterion = criterion.eval()
+    criterion = criterion.to(device)
+
+    if lr_scheduler is None:
+        if get_optimizer is None:
+            get_optimizer = ulyanov_et_al_2016_optimizer
+        optimizer = get_optimizer(transformer)
+        lr_scheduler = ulyanov_et_al_2018_lr_scheduler(
+            optimizer, impl_params=impl_params,
+        )
+
+    if num_epochs is None:
+        if impl_params:
+            if instance_norm:
+                num_epochs = 25
+            else:
+                num_epochs = 10 if stylization else 5
+        else:
+            num_epochs = 10
+
+    style_transform = ulyanov_et_al_2016_style_transform(
+        impl_params=impl_params, instance_norm=instance_norm
+    )
+    style_transform = style_transform.to(device)
+    style_image = style_transform(style_image)
+    style_image = batch_up_image(style_image, loader=content_image_loader)
+
+    criterion.set_style_image(style_image)
+
+    def criterion_update_fn(input_image, criterion, preprocessor=None):
+        if hasattr(criterion, "content_loss"):
+            if preprocessor is None:
+                preprocessor = ulyanov_et_al_2016_preprocessor()
+            preprocessor = preprocessor.to(device)
+            criterion.set_content_image(preprocessor(input_image))
+
+    default_transformer_epoch_optim_loop(
+        content_image_loader,
+        transformer,
+        criterion,
+        criterion_update_fn,
+        num_epochs,
+        device=device,
+        lr_scheduler=lr_scheduler,
+        quiet=quiet,
+        logger=logger,
+        log_fn=log_fn,
+    )
+
+    return transformer
+
+
+def ulyanov_et_al_2016_stylization(
+    input_image: torch.Tensor,
+    transformer: Union[nn.Module, str],
+    impl_params: bool = True,
+    instance_norm: bool = None,
+    sample_size: int = 256,
+    postprocessor: ulyanov_et_al_2016_postprocessor = None,
+):
+    device = input_image.device
+    if isinstance(transformer, str):
+        style = transformer
+        transformer = ulyanov_et_al_2016_transformer(
+            style=style, impl_params=impl_params, instance_norm=instance_norm,
+        )
+        transformer = transformer.eval()
```

Comment:
He uses `model:evaluate()` only in his master branch. In `texture_nets_v1` he does not set it to evaluation mode before the stylization process or texture sampling. I made it conditional so that this step is only skipped in the branch `texture_nets_v1`.